### PR TITLE
NAS-128574 / 24.10 / Allow HeaderDigest and DataDigest to be configured for iSCSI targets

### PR DESCRIPTION
**Pull request by bmeagherix**
Enhance iSCSI APIs to allow some target options to be specified

### CODE REVIEW

File: src/middlewared/middlewared/plugins/iscsi_/targets.py [hunk 1]
```diff
@@ -97,6 +127,18 @@ async def extend(self, data):
             ),
         ]),
         List('auth_networks', items=[IPAddr('ip', network=True)]),
+        Dict(
+            'options',
+            Str('DataDigest', enum=DigestEnum.choices()),
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/iscsi_/targets.py:132]: I don't think we typically have camelcase parameters. Convention is all lower-case with underscores as-needed.

Comment (bmeagherix) [reply src/middlewared/middlewared/plugins/iscsi_/targets.py:132]: I understand, but we're exposing **raw** iSCSI settings, so IMO this is clearer.

e.g. https://www.rfc-editor.org/rfc/rfc7143#section-13.1

Comment (anodos325) [reply src/middlewared/middlewared/plugins/iscsi_/targets.py:132]: @yocalebo do you have strong opinion on this one?

**Comment by bmeagherix** [comment]
Closing for now.  Need to perform performance analysis before proceeding.

**Pull request closed without merging**